### PR TITLE
Added More Borders

### PR DESCRIPTION
**Pull request by zacharias1219**
Issue no #500

### CODE REVIEW

File: mesop/component_helpers/style.py [hunk 1]
```diff
@@ -73,12 +73,26 @@ class BorderSide:
                                 a string with a unit (e.g., '2em'), or None for no width.
       color: The color of the border, represented as a string. This can be any valid CSS color value,
                           or None for no color.
-      style: The style of the border, which can be 'none' for no border, 'solid' for a solid line.
+      style: The style of the border.
```
Comment (richard-to) [inline mesop/component_helpers/style.py:76]: Can you update the comment to also include a link where the types of border styles are. Usually we use developer.mozilla.org links: https://developer.mozilla.org/en-US/docs/Web/CSS/border-style

File: tools/defaults.bzl [hunk 2]
```diff
@@ -364,17 +364,17 @@ def ng_web_test_suite(browsers = [], deps = [], static_css = [], exclude_init_sc
             srcs = [css_label],
             outs = ["%s.css.js" % css_id],
             output_to_bindir = True,
-            cmd = """
```
Comment (richard-to) [inline tools/defaults.bzl:367]: Hmm, this file seems accidentally updated.

**Review by richard-to — CHANGES_REQUESTED** [review]
Can you also update one of the demo examples (Maybe the "box" demo) to illustrate different border styles.

**Comment by richard-to** [comment]
Hey @zacharias1219 

Also looks like someone else also posted a pull request for the border side change before you: https://github.com/google/mesop/pull/504. Looks like what happened is that they posted message saying they would work on it after I posted a message for you to look at this one. Sorry about that.

Since they posted a message first volunteering, I'm going to give their pull request precedence. If their pull request gets abandoned I will merge yours in. So for now hold off on making adjustments.

Again, sorry about that. I really appreciate that you want to help out with Mesop. We'll have to figure out a better way of assigning issues since this isn't the first time we got duplicate pull requests.

**Comment by zacharias1219** [comment]
Ahh that's fine, no worries, plenty more issues for me to work on.

**Comment by richard-to** [comment]
Looks like the other person may have stopped with their Pull Request for this fix. If you're still up for it, let's try to your fix merged in instead.

File: mesop/component_helpers/style.py (hunk 1 above)
Comment (zacharias1219) [reply mesop/component_helpers/style.py:76]: Hey, so should i just add that link to the comment?
I've committed the file to just see if all the checks pass, I'll make these changes next.

**Comment by richard-to** [comment]
Yes I think a link is fine

**Comment by zacharias1219** [comment]
Hey so I've added the link in the comments.
Hopefully this is fine.

**Comment by richard-to** [comment]
Change seem fine. Can you update demo/box.py to include some example usages of the border styles.

**Comment by zacharias1219** [comment]
Sure, I've made some changes, here you go
`with me.box(
        style=me.Style(
            background="blue",
            height=50,
            margin=me.Margin.all(16),
            border=me.Border.all(
                me.BorderSide(width=2, color="yellow", style="dotted")
            ),
            border_radius=10,
        )
    ):
        me.text(text="Example with all sides bordered")

    with me.box(
        style=me.Style(
            background="purple",
            height=50,
            margin=me.Margin.symmetric(vertical=24, horizontal=12),
            border=me.Border(
                top=me.BorderSide(width=4, color="white", style="double"),
                bottom=me.BorderSide(width=4, color="white", style="double"),
            ),
        )
    ):
        me.text(text="Example with top and bottom borders")

    with me.box(
        style=me.Style(
            background="cyan",
            height=50,
            margin=me.Margin.symmetric(vertical=24, horizontal=12),
            border=me.Border(
                left=me.BorderSide(width=2, color="black", style="groove"),
                right=me.BorderSide(width=2, color="black", style="groove"),
            ),
        )
    ):
        me.text(text="Example with left and right borders")`

File: demo/box.py [hunk 3]
```diff
@@ -22,3 +22,42 @@ def app():
...
+      )
+    ):
+      me.text(text="Example with all sides bordered")
+
+    with me.box(
+      style=me.Style(
+        background="purple",
+        height=50,
+        margin=me.Margin.symmetric(vertical=24, horizontal=12),
+        border=me.Border(
```
Comment (richard-to) [inline demo/box.py:44]: [nit] Border has a static method called symmetric. So you can do a shortcut:

```
me.Border.symmetric(vertical==me.BorderSide(width=4, color="white", style="double"))
```

File: demo/box.py [hunk 4]
```diff
@@ -22,3 +22,42 @@ def app():
...
+      )
+    ):
+      me.text(text="Example with top and bottom borders")
+
+    with me.box(
+      style=me.Style(
+        background="cyan",
+        height=50,
+        margin=me.Margin.symmetric(vertical=24, horizontal=12),
+        border=me.Border(
```
Comment (richard-to) [inline demo/box.py:57]: [nit] Border has a static method called symmetric. So you can do a shortcut:

```
me.Border.symmetric(horizontal=me.BorderSide(width=2, color="black", style="groove"))
```

File: mesop/component_helpers/style.py [hunk 5]
```diff
@@ -73,12 +73,27 @@ class BorderSide:
...
+      https://developer.mozilla.org/en-US/docs/Web/CSS/border-style
```
Comment (richard-to) [inline mesop/component_helpers/style.py:77]: [nit] I think this should go on the same line as the style arg.

```
The style of the border. See https://developer.mozilla.org/en-US/docs/Web/CSS/border-style
```

**Review by richard-to — CHANGES_REQUESTED** [review]
This looks good overall. I had a few more minor clean up comments. And I think that should be it.

File: mesop/component_helpers/style.py (hunk 5 above)
Comment (zacharias1219) [reply mesop/component_helpers/style.py:77]: Sure will do the change

File: demo/box.py (hunk 4 above)
Comment (zacharias1219) [reply demo/box.py:57]: Basicalyl did this
`
  with me.box(
      style=me.Style(
        background="cyan",
        height=50,
        margin=me.Margin.symmetric(vertical=24, horizontal=12),
        border=me.Border.symmetric(horizontal=me.BorderSide(width=2, color="black", style="groove"))
      )
    ):
      me.text(text="Example with left and right borders")
`

File: demo/box.py (hunk 3 above)
Comment (zacharias1219) [reply demo/box.py:44]: Here it is saying vertical is not defined

Comment (zacharias1219) [reply demo/box.py:44]: Wait fixed it

**Review by richard-to — APPROVED** [review]

**Comment by richard-to** [comment]
Awesome. All the tests passed. Thanks for helping with this change!

**Comment by zacharias1219** [comment]
Ahh No problem, I have semester exams coming up, so my activity might decrease, but will surely try to contribute more.
I'll be deleting this branch.

**Comment by richard-to** [comment]
Good luck on your semester exams. And for sure, you should definitely focus on your school work. Contributing to Mesop, I would say is low priority. Definitely only do so if you feel like it.

**Pull request merged**